packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -67,25 +118,10 @@ export class TreeActions {
   private loadSubTree(options: LoadNodesOptions, initialRootNode?: TreeModelRootNode) {
     const loadAction = this.getLoadAction(options.parent.id);
     const timeTracker = new TimeTracker((time) => this._onLoad(loadAction, time));
-    const parentId = options.parent.id;
-    this._loader
-      .loadNodes(options)
-      .pipe(collectTreePartsUntil(this._reset, initialRootNode))
-      .subscribe({
-        next: (newModel) => {
-          const childNodes = newModel.parentChildMap.get(parentId);
-          const firstChildNode = childNodes?.length ? newModel.idToNode.get(childNodes[0]) : undefined;
-          this.handleLoadedHierarchy(parentId, newModel);
-          // only report load duration if no error occurs
-          if (!(firstChildNode && isTreeModelInfoNode(firstChildNode))) {
-            timeTracker.finish();
-          }
-        },
-        complete: () => {
-          this.onLoadingComplete(parentId);
-          timeTracker[Symbol.dispose]();
-        },
-      });
+    const isComplete = new Subject<void>();
```

Comment:
Maybe it would make more sense to return a `Promise<void>` for this? Then, in the tests we could do `await actions.reloadTree();`, which looks cleaner than this:

```ts
const isComplete = actions.expandNode("root-1", true);
await waitFor(() => {
  expect(isComplete?.closed).to.be.true;
});
```

The implementation wouldn't get more complicated either:

```ts
return new Promise((resolve) => {
    // the loader should call `onComplete` callback instead of subscribing to and unsubscribing from `isComplete` subject
    this._nodeLoader.next({ loadOptions: options, onComplete: resolve, timeTracker, parentId: options.parent.id, initialRootNode });
});
```

I think that would make tests easier to read.

Reply:
I do agree it would be easier to read but all the functions that return this would need to become async.

Reply:
> We'd need to add void to all the reload calls, I don't know how I feel about that.

Yeah, that's not great. What if we return `{ completed: Promise<void> }` object? We could still `await reload().complete`, but simply calling `reload()` wouldn't require an await.

Reply:
I think that works, changed.